docs/docs/api/resource-management.md

```diff
@@ -35,14 +35,18 @@ this would lose a lot of the benefits of using asynchronity and we might almost
 We can do better, retaining concurrency and not leaking resources, by using `.using`:
 
 ```js
-var using = Promise.using;
-
-using(getConnection(),
-      fs.readFileAsync("file.sql", "utf8"), function(connection, fileContents) {
-    return connection.query(fileContents);
-}).then(function() {
-    console.log("query successful and connection closed");
-});
+function doStuff() {
+    return Promise.using([
+        connectionPool.getConnectionAsync(),
+        fs.readFileAsync("file.sql", "utf8")
+    ]), function(connection, fileContents) {
```

Comment:
The array format doesn't quite work that way. It will pass a single argument to the function - an array of allocated resources. 

Its the intention that the array variant should be used with an array of (same or similar) resources. Bluebird's `Promise.using` is actually more similar to `Promise.join`. I'd recommend using the non-array variant for a fixed number of heterogenous resources (one of the reasons being that its much easier to write type definitions for those).